mmv1/templates/terraform/encoders/spanner_instance_update.go.tmpl

```diff
@@ -21,32 +21,36 @@ if d.HasChange("labels") {
-	oldSlice := old.([]interface{})
-	newSlice := new.([]interface{})
-	if len(oldSlice) == 0 || len(newSlice) == 0 {
-		updateMask = append(updateMask, "autoscalingConfig")
-	} else {
-		if d.HasChange("autoscaling_config.0.autoscaling_limits.0.max_processing_units") {
-			updateMask = append(updateMask, "autoscalingConfig.autoscalingLimits.maxProcessingUnits")
-		}
-		if d.HasChange("autoscaling_config.0.autoscaling_limits.0.min_processing_units") {
-			updateMask = append(updateMask, "autoscalingConfig.autoscalingLimits.minProcessingUnits")
-		}
-		if d.HasChange("autoscaling_config.0.autoscaling_limits.0.max_nodes") {
-			updateMask = append(updateMask, "autoscalingConfig.autoscalingLimits.maxNodes")
-		}
-		if d.HasChange("autoscaling_config.0.autoscaling_limits.0.min_nodes") {
-			updateMask = append(updateMask, "autoscalingConfig.autoscalingLimits.minNodes")
-		}
-		if d.HasChange("autoscaling_config.0.autoscaling_targets.0.high_priority_cpu_utilization_percent") {
-			updateMask = append(updateMask, "autoscalingConfig.autoscalingTargets.highPriorityCpuUtilizationPercent")
-		}
-		if d.HasChange("autoscaling_config.0.autoscaling_targets.0.storage_utilization_percent") {
-			updateMask = append(updateMask, "autoscalingConfig.autoscalingTargets.storageUtilizationPercent")
-		}
-	}
+    old, new := d.GetChange("autoscaling_config")
+    oldSlice := old.([]interface{})
+    newSlice := new.([]interface{})
+    if len(oldSlice) == 0 || len(newSlice) == 0 {
+        updateMask = append(updateMask, "autoscalingConfig")
+    } else {
+        if d.HasChange("autoscaling_config.0.autoscaling_limits.0.max_processing_units") {
+            updateMask = append(updateMask, "autoscalingConfig.autoscalingLimits.maxProcessingUnits")
+        }
+        if d.HasChange("autoscaling_config.0.autoscaling_limits.0.min_processing_units") {
+            updateMask = append(updateMask, "autoscalingConfig.autoscalingLimits.minProcessingUnits")
+        }
+        if d.HasChange("autoscaling_config.0.autoscaling_limits.0.max_nodes") {
+            updateMask = append(updateMask, "autoscalingConfig.autoscalingLimits.maxNodes")
+        }
+        if d.HasChange("autoscaling_config.0.autoscaling_limits.0.min_nodes") {
+            updateMask = append(updateMask, "autoscalingConfig.autoscalingLimits.minNodes")
+        }
+        if d.HasChange("autoscaling_config.0.autoscaling_targets.0.high_priority_cpu_utilization_percent") {
+            updateMask = append(updateMask, "autoscalingConfig.autoscalingTargets.highPriorityCpuUtilizationPercent")
+        }
+        if d.HasChange("autoscaling_config.0.autoscaling_targets.0.storage_utilization_percent") {
+            updateMask = append(updateMask, "autoscalingConfig.autoscalingTargets.storageUtilizationPercent")
+        }
+        if d.HasChange("autoscaling_config.0.asymmetric_autoscaling_options") {
+            updateMask = append(updateMask, "autoscalingConfig.asymmetricAutoscalingOptions")
+        }
+    }
```

Comment:
only nit: Let's switch to tabs for indentation to maintain consistency and avoid showing unrelated changes in the diff and change history.